Gems/AtomLyIntegration/CommonFeatures/Code/Source/Mesh/MeshComponentController.cpp

```diff
@@ -699,6 +701,74 @@ namespace AZ
             return Aabb::CreateNull();
         }
 
+        void MeshComponentController::GetVisibleGeometry(AzFramework::VisibleGeometryContainer& geometryContainer) const
+        {
+            // Attempt to copy the triangle list geometry data out of the model asset into the visible geometry structure
+            const auto& modelAsset = GetModelAsset();
+            if (!modelAsset.IsReady() || modelAsset->GetLodAssets().empty())
+            {
+                return;
+            }
+
+            // This will only extract data from the first LOD. It might be necessary to make the LOD selectable.
+            const auto& lodAsset = modelAsset->GetLodAssets().front();
+            if (!lodAsset.IsReady())
+            {
+                return;
+            }
+
+            const AZ::Name positionName{ "POSITION" };
+            for (const auto& mesh : lodAsset->GetMeshes())
+            {
+                // Get the index buffer data, confirming that the asset is valid and indices are 32 bit integers. Other formats are
+                // currently not supported.
+                const AZ::RPI::BufferAssetView& indexBufferView = mesh.GetIndexBufferAssetView();
+                const AZ::RHI::BufferViewDescriptor& indexBufferViewDesc = indexBufferView.GetBufferViewDescriptor();
+                const AZ::Data::Asset<AZ::RPI::BufferAsset> indexBufferAsset = indexBufferView.GetBufferAsset();
+                if (!indexBufferAsset.IsReady() || indexBufferViewDesc.m_elementSize != sizeof(uint32_t))
+                {
+                    continue;
+                }
+
+                // Get the position buffer data, if it exists with the expected name.
+                const AZ::RPI::BufferAssetView* positionBufferView = mesh.GetSemanticBufferAssetView(positionName);
+                if (!positionBufferView)
+                {
+                    continue;
+                }
+
+                // Confirm that the position buffer is valid and contains 3 32 bit floats for each position. Other formats are currently not
+                // supported.
+                const AZ::RHI::BufferViewDescriptor& positionBufferViewDesc = positionBufferView->GetBufferViewDescriptor();
+                const AZ::Data::Asset<AZ::RPI::BufferAsset> positionBufferAsset = positionBufferView->GetBufferAsset();
+                if (!positionBufferAsset.IsReady() || positionBufferViewDesc.m_elementSize != sizeof(float) * 3)
```

Comment:
The XYZ coordinates being represented by the value of 3 should be represented a by constant(or a setting that defaults to 3)
Such as `constexpr uint32_t AxisPerVertex = 3;`